src/Symfony/Component/Uid/Uuid.php

```diff
@@ -214,6 +217,11 @@ private static function transformToRfc4122(string $uuid, int $format): string
             $uuid = $ulid->toRfc4122();
         }
 
+        if ($inputUuid === $uuid && !($format & self::FORMAT_RFC_4122)) {
+            // input format doesn't match the input string
+            return false;
```

Comment:
Why not throw here ?

Reply:
I didn't want to trigger the heavy exception mechanism (even more for an exception we don't rethrow to the user) for something that we want to be very fast, like validating an UUID